### PR TITLE
Specialized StreamField blocks and typed chooser panel

### LINKED ISSUE #133: Add type filter to MediaChooserPanel
so one can do `MediaChooserPanel("field_name", type="video")` to only allow videos to be selected

Alternatively, provide separate `VideoChooserPanel` and `AudioChooserPanel`s

### PR DESCRIPTION
**Pull request by zerolab**
This PR fixes #133

To-Do:
- [x]  update README

### CODE REVIEW

File: src/wagtailmedia/widgets.py [hunk 1]
```diff
@@ -34,9 +34,23 @@ class AdminMediaChooser(AdminChooser):
     link_to_chosen_text = _("Edit this media item")
 
     def __init__(self, **kwargs):
-        super(AdminMediaChooser, self).__init__(**kwargs)
+        media_type = kwargs.pop("media_type", None)
+        super().__init__(**kwargs)
         self.media_model = get_media_model()
 
+        if media_type == "audio":
```
Comment (kaedroho) [inline src/wagtailmedia/widgets.py:41]: Is it possible to create sub-classes for each media type, for consistency with blocks?

Also, do we want to be able to support custom media types? (such as svg, 3d models, etc)

File: tests/test_blocks.py [hunk 2]
```diff
@@ -12,27 +16,96 @@ def setUpTestData(cls):
...
+
+        # note: can't use self.assertQuerysetEqual() as it is funky in Django < 3.2
+        self.assertListEqual(
+            list(block.field.queryset.order_by("pk").values_list("pk", flat=True)),
+            list(Media.objects.order_by("pk").values_list("pk", flat=True)),
+        )
+
+        block = AbstractMediaChooserBlock(media_type="audio")
+        self.assertListEqual(
+            list(block.field.queryset.values_list("pk", flat=True)),
```
Comment (kaedroho) [inline tests/test_blocks.py:62]: This assertion is missing the `order_by`, does this matter?

Also applies to:
- ``test_audio_chooser_block_field_queryset``
- ``test_video_chooser_block_field_queryset``

File: tests/test_edit_handlers.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+class MediaChooserPanelTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.request = RequestFactory().get("/")
+        cls.request.user = (
+            AnonymousUser()
+        )  # technically, Anonymous users cannot access the admin
+
+        fake_file = ContentFile("Test")
+        fake_file.name = "test.mp3"
```
Comment (kaedroho) [inline tests/test_edit_handlers.py:24]: Do we need some kind of file type validation?

File: tests/testapp/models.py [hunk 4]
```diff
@@ -114,11 +118,20 @@ class BlogStreamPage(Page):
...
+    featured_media = models.ForeignKey(
+        "wagtailmedia.Media", on_delete=models.PROTECT, related_name="+"
+    )
+
     content_panels = Page.content_panels + [
         FieldPanel("author"),
         FieldPanel("date"),
         StreamFieldPanel("body"),
+        MediaChooserPanel("featured_media"),
+        # MediaChooserPanel("featured_media", media_type="audio"),
```
Comment (kaedroho) [inline tests/testapp/models.py:135]: Are these needed? Perhaps we could create separate fields for them?

**Review by kaedroho — COMMENTED** [review]
Nice work! Just some minor comments

File: tests/test_blocks.py (hunk 2 above)
Comment (zerolab) [reply tests/test_blocks.py:62]: in this case it doesn't matter as both return 1 item. it fails for > 1 results

File: tests/testapp/models.py (hunk 4 above)
Comment (zerolab) [reply tests/testapp/models.py:135]: I left them for convenience when checking the interface via `tox -e interactive`
had separate ones but thought that is too much?

File: src/wagtailmedia/widgets.py (hunk 1 above)
Comment (zerolab) [reply src/wagtailmedia/widgets.py:41]: good suggestion! I did the split for the blocks quite late, but creating custom widget classes means we'd have a pattern for other media types, should we want to support them

Comment (zerolab) [reply src/wagtailmedia/widgets.py]: done now

**Pull request merged**